### PR TITLE
Components: Add <Card /> tests

### LINKED ISSUE #7939: Components: Add tests for <Card /> component
The [`<Card />` component](https://github.com/Automattic/wp-calypso/tree/master/client/components/card) is currently not tested. We should seek to add tests to verify expected render behavior of the component. [Enzyme](http://airbnb.io/enzyme/) (and [`chai-enzyme` assertions](https://github.com/producthunt/chai-enzyme)) may be useful utilities for rendering and asserting expected render results. Ideas for test cases may include some or all of the following:

- Render returns expected element
 - Optionally rendered components are or are not present
 - Modifier classes by incoming props or component state
- User interactions have expected results
- Impact of lifecycle methods when props or state change during lifetime of component

When submitting a pull request, reference this issue.

### PR DESCRIPTION
**Pull request by scottsweb**
Introduce tests for the <Card/> component. Tests include:

**Card**

* should have `card` class
* should have custom class of `test__ace`
* should render children
* should be linkable

**CompactCard**

* should have `is-compact` class
* should have custom class of `test__ace`
* should render children
* should use the card component

Closes #7939

To run these tests use:

```
npm run test-client client/components/card
```

cc @jeremeylduvall

### CODE REVIEW

File: client/components/card/test/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+
+//enzyme  - simulate
```
Comment (bluefuton) [inline client/components/card/test/index.js]: Is this needed?

File: client/components/card/test/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * Internal dependencies
+ */
+const Card = require( '../' );
```
Comment (bluefuton) [inline client/components/card/test/index.js]: Can we use `import` here instead?

**Comment by scottsweb** [comment]
Thanks for reviewing @bluefuton - I have addressed those two issues. I think we are good to go 👍

File: client/components/card/test/index.js (hunk 2 above)
Comment (scottsweb) [inline client/components/card/test/index.js]: Sure. Done.

**Comment by bluefuton** [comment]
Nice one. 🚢 away!

File: client/components/card/test/index.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	// it should have a class of `card`
+	it( 'should have `card` class', function() {
+		const card = shallow( <Card /> );
+		expect( card.is( '.card' ) ).to.equal( true );
+	} );
+
+	// it should accept a custom class of `test__ace`
+	it( 'should have custom class of `test__ace`', function() {
+		const card = shallow( <Card className="test__ace" /> );
+		expect( card.is( '.card' ) ).to.equal( true );
```
Comment (bluefuton) [inline client/components/card/test/index.js]: @scottsweb just spotted that this should be:

`expect( card.is( '.test__ace' ) ).to.equal( true );`

Comment (bluefuton) [inline client/components/card/test/index.js]: Fixed in https://github.com/Automattic/wp-calypso/pull/8871.

**Pull request merged**